frontend/src/components/studySamples/StudyStepSamplesTable.tsx

```diff
@@ -116,14 +102,16 @@ function StudyStepSamplesTable({ studyID, step, settings }: StudyStepSamplesTabl
 		<WorkflowSamplesTable
 			clearFilters={localClearFilters}
 			hasFilter={true}
-			samples={samples ?? []}
+			samples={step.ready.samples}
 			columns={columns}
 			filterDefinitions={{ ...SAMPLE_COLUMN_FILTERS, ...LIBRARY_COLUMN_FILTERS }}
 			filterKeys={{ ...SAMPLE_NEXT_STEP_BY_STUDY_FILTER_KEYS, ...SAMPLE_NEXT_STEP_BY_STUDY_LIBRARY_FILTER_KEYS }}
 			filters={settings?.filters ?? {}}
 			setFilter={setFilter}
 			setFilterOptions={setFilterOptions}
 			setSortBy={setSortBy}
+			pagination={{ pageNumber, pageSize, totalCount: step.ready.count, onChangePageNumber, onChangePageSize }}
+			loading={tableState?.isFetching}
```

Comment:
test tableState and send either isFetching or true instead in order not to make loading optional ?

Reply:
```ts
			loading={tableState?.isFetching ?? true}
```
?